### ISSUE TITLE
Window Dark Mode Improvements

### ISSUE THREAD

**Issue by nlogozzo** [moneymanagerex/moneymanagerex#4811]
I'm a developer using wxWidgets who's trying to provide a good dark mode experience for Windows users. 
I noticed how your application handles dark mode and was thinking that it could use some improvement using Win32 calls. 
I created a function in my application:
```cpp
#if (defined (_WIN32) || defined (_WIN64))
#include <dwmapi.h>
#endif

void ThemeHelpers::ApplyWin32DarkMode(wxWindow* window)
{
#if (defined (_WIN32) || defined (_WIN64))
  BOOL enabled = TRUE;
  DwmSetWindowAttribute(window->GetHWND(), 20, &enabled, sizeof(enabled));
  SetWindowTheme(window->GetHWND(), L"DarkMode_Explorer", NULL);
#endif
}
```
The result is a dark title bar and dark scroll bars throughout the application.
![image](https://user-images.githubusercontent.com/17648453/178187872-cff4c2c5-e7a1-4065-8a9a-0e63ed5cb6e0.png)
This works on Windows 10 & 11.
The only control I cannot seem to apply a dark color to is the MenuBar. And it looks like no one else has had any other luck either.

**Comment by vomikan** [comment]
https://github.com/moneymanagerex/moneymanagerex/issues/4545

**Comment by n-stein** [comment]
While checking out #5861 I found that wxWidgets 3.3.0 will give us a significant improvement in Windows Dark Mode just by adding a single line to mmex.cpp:

`app->MSWEnableDarkMode();`

This applies dark mode to all title bars, menus, scroll bars, panel separators, and even the App Start dialog.

Before:
![image](https://user-images.githubusercontent.com/16599929/234759148-8f06300b-12c7-4bba-8324-dea1b98f5267.png)
![image](https://user-images.githubusercontent.com/16599929/234759159-cb4886f0-99ce-41db-9796-816b34586809.png)

After:
![image](https://user-images.githubusercontent.com/16599929/234759323-9023510e-6d74-44ae-98c0-88017184dd04.png)
![image](https://user-images.githubusercontent.com/16599929/234759364-d8c04645-4daf-47f4-aa9f-375fc448e8d1.png)

This is not yet released and there are a few issues being worked out such as text color on certain controls. Just thought I'd share an update.

**Comment by vomikan** [comment]
https://github.com/moneymanagerex/moneymanagerex/issues/4545

Seems dark mode does not working on Windows for help pages.

**Comment by n-stein** [comment]
Yeah, no change there. The issue there is the default WebView backend uses IE which doesn't support the prefers-color-scheme media query.

![image](https://user-images.githubusercontent.com/16599929/235226221-9e7e4509-07dd-4f25-9019-5a44fd95709c.png)


I've compiled wxWidgets with MS Edge WebView2 as the default backend (wxUSE_WEBVIEW_EDGE), and that does fix the help page:
![image](https://user-images.githubusercontent.com/16599929/235258621-7529b7dd-cf54-4e5d-9315-b36f52bd72ff.png)

But need to figure out how to make reports work with Edge. They currently don't render properly, probably due to missing custom schema and VFS support:
![image](https://user-images.githubusercontent.com/16599929/235259046-b746e8c3-e489-48fa-a272-4dbafedde9ae.png)

**Comment by n-stein** [comment]
#4905

**Comment by ovari** [comment]
![image](https://github.com/moneymanagerex/moneymanagerex/assets/17465872/b17684f7-a987-46b7-8bec-a5d5c4ec32f3)

```
Money Manager Ex
• Version: 1.8.0-RC.2 64-bit
• Built on: Apr 26 2024 03:16:58
• Database version: 19 (aes128cbc)
• Git commit: 99b397813 (2024-04-25)
• Git branch: master
MMEX is using the following support products:
• wxWidgets 3.2.4 (wxGTK 3.24)
• wxSQLite3 4.9.1 (SQLite 3.40.0)
• RapidJSON 1.1.0
• Lua 5.3.5
• libcurl/8.7.0-DEV OpenSSL/3.1.5 zlib/1.3.1 libidn2/2.3.4 libpsl/0.21.2 nghttp2/1.58.0
• gettext 0.22
• apexcharts.js
Built with:
• CMake 3.29.2
• GNU Make 4.4.1
• GCC 13.2.0
Running on:
• GNOME.45 (Flatpak runtime)
• GNOME 45 (Flatpak runtime) ""
• Linux 5.15.0-105-generic x86_64
• X-Cinnamon (X11)
• System Locale: en_AU.UTF-8
• User Interface Language: en_GB.UTF-8
• 1368x768 24-bit 96x96ppi
```

**Comment by n-stein** [comment]
@ovari please create a new issue. This issue is Windows specific with a dependency on wxWidgets.
